content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,846 @@
+    float m_bendingStrength;
+    float m_windBendingFrequency;
+    ```
+
+1. Open the leaf material (`aspen_leaf.material`) in the **Material Editor**. Make sure you select the leaf material because that is the parent material of the other parts of the tree.
+
+1. In the Material Editor **Inspector**, find **Vegetation Bending** and ensure that the 6 properties you just added are there.
+
+1. Adjust the properties! Make sure you adjust all of them so that you can see bending in the later steps.
+
+### Add process bending function
+First, let's add a function that your multiple vertex shaders can call to avoid repeat code. You will be writing more functions for different parts of the bending and calling them all from this function. 
+1. Open `VegetationBending_Common.azsli`.
+
+1. At the bottom, add a function to return the world position of the vertex after applying bending. The parameters given to this function are all helpful for determining bending.
+
+   ```hlsl
+   float4 ProcessBending(float currentTime, float3 objectSpacePosition, float3 normal, float4 detailBendingParams, float4 worldPosition, float4x4 objectToWorld) 
+   {
+      float4 adjustedWorldPosition = float4(worldPosition);
+      if (o_color_isBound) 
+      {
+         // You will add function calls here.
+      }
+      return adjustedWorldPosition;
+   }
+   ```
+
+   {{< note >}}
+   Notice how, like before, you are using a conditional with `o_color_isBound` to ensure that only meshes with vertex streams perform bending.
+   {{< /note >}}
+
+1. Call the `ProcessBending` function in your vertex shaders. Open `VegetationBending_ForwardPass.azsli` and find the vertex shader (`VegetationBending_ForwardPassVS`). Add a call to your function above `OUT.m_worldPosition = worldPosition.xyz`. Most of the parameters to pass in are inputs to your vertex shader or values you have calculated already, and you can get the time from the property `m_time` the *scene shader resource group* (`SceneSrg`), which gives us the number of seconds since the start of the application.
+   
+   ```hlsl
+   float currentTime = SceneSrg::m_time;
+   worldPosition = ProcessBending(currentTime, IN.m_position, IN.m_normal, IN.m_optional_color, worldPosition, objectToWorld);
+
+   OUT.m_worldPosition = worldPosition.xyz;
+   OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, worldPosition);
+
+   return OUT;
+   ``` 
+   {{< note >}}
+   The code needs to be above the two `OUT` lines because you will edit `worldPosition`, which adjusts the `OUT` variables accordingly.
+   {{< /note >}}
+  
+1. Repeat steps 2-3 with the depth pass in `VegetationBending_DepthPass.azsli` and the depth pass with PS in `VegetationBending_DepthPass_WithPS.azsli`.
+
+   {{< tip >}}
+   Note that the depth pass does not output a world position. Just place the code above the `OUT.m_position` line. The depth pass with PS has some extra code in the vertex shader, but you can still place this code about the two `OUT` statements.
```

Comment:
If there's no special instructions for the depth pass with PS, I suggest removing it completely.
```suggestion
   For  `VegetationBending_DepthPass.azsli`, you only need to place the code above the `OUT.m_position` line because the depth pass does not output a world position
```